src/v/kafka/server/handlers/fetch.h

```diff
@@ -359,9 +360,11 @@ struct fetch_plan {
     explicit fetch_plan(
       size_t shards,
       op_context::latency_point start_time = op_context::latency_clock::now())
-      : fetches_per_shard(shards, shard_fetch(start_time)) {
-        for (size_t i = 0; i < fetches_per_shard.size(); i++) {
-            fetches_per_shard[i].shard = i;
+      : fetches_per_shard() {
+        fetches_per_shard.reserve(shards);
+        for (size_t i = 0; i < shards; i++) {
+            auto& fps = fetches_per_shard.emplace_back(start_time);
```

Comment:
IMO it's better to just change the `shard_fetch` ctor to take time & shard now, and then you can pass them both to emplace and void this weird thing where we fill in the shard after (should be no slower, may be faster). Then we don't leave shard uninit any more which is good.

I believe it was me who did it like this, mostly just to avail myself of the "n copies" constructor but we aren't using that any more.

Reply:
(approving though so feel free to ignore)

Reply:
done